client/src/plugins/camunda-plugin/deployment-tool/DeploymentConfigModal.js

```diff
@@ -114,7 +118,32 @@ export default class DeploymentConfigModal extends React.PureComponent {
 
   onClose = (action = 'cancel', data) => this.props.onClose(action, data);
 
-  onCancel = () => this.onClose('cancel');
+  onCancelButtonPressed = () => {
+
+    const {
+      onClose,
+      initialConfigurations
+    } = this;
+
+    const {
+      saveCredential,
+      removeCredentials
+    } = this.props;
+
+    // we want to revert authentication related configurations
+    // if the user presses on Cancel button
+    const {
+      username, password, token, rememberCredentials
+    } = initialConfigurations.endpoint;
+
+    if (!rememberCredentials) {
+      removeCredentials();
+    } else {
+      saveCredential({ username, password, token });
+    }
```

Comment:
How about if we really save the credentials only when the `Deploy` button is clicked? Then, we should also be able to remove the save logic from the render part (e.g. https://github.com/camunda/camunda-modeler/pull/1721/files#diff-93a838d26a593c08ab9930e4e2b4ae15R411).

So instead of doing:
initial values => <user types username> => save username  => <user clicks cancel> => save initial username => close
we would do this:
initial values => <user types username> => <user clicks cancel> => close

If you want to discuss it or do a remote pair programming, feel free to reach out to me! :)

Reply:
Actually, the `remeberCredentials` logic is [already there](https://github.com/camunda/camunda-modeler/blob/c1508e72bac3ef21149fae2ba226c83d50d6a1e5/client/src/plugins/camunda-plugin/deployment-tool/DeploymentTool.js#L237). I would suggest then to simply remove the code where we save credentials on field value change.

Reply:
By design we don't save credentials only when the Deploy is clicked, so it's up to our designers choice. I'm not touching there :) (not-my-job)

Reply:
We also save when X is clicked as well

Reply:
We could do -> [User Clicks on Deploy] || [User Clicks on X] => Save though

Reply:
I missed this part:
> X for closing the view (with the side effect of saving intermediate work)

Still, we can save when the X button is actually clicked and not for each change in the related fields.

Reply:
> We could do -> [User Clicks on Deploy] || [User Clicks on X] => Save though

Exactly :)

Reply:
Yeah I think I'll simplify here.

Reply:
Done -> https://github.com/camunda/camunda-modeler/pull/1721/commits/42791b5ebd5d56fa7f7dbd645319996c54779e28